source/agora/consensus/state/UTXOSet.d

```diff
@@ -49,31 +49,27 @@ abstract class UTXOCache
         Params:
             tx = the transaction
             height = local height of the block
-            slashed = whether the transaction is for slashing
+            commons_budget = the CommonsBudget address
 
     ***************************************************************************/
 
     public void updateUTXOCache (const ref Transaction tx, Height height,
-        bool slashed = false) @safe
+        immutable PublicKey commons_budget) @safe
     {
         import std.algorithm : any;
 
-        // defaults to next block
```

Comment:
Do you really need to introduce a new variable here? I think you can still use `unlock_height`.

Just set it to `height + 2016` for all frozen utxos and do something like;

``` d
auto utxo_value = UTXO(output.address != commons_budget ? 
           unlock_height : height + 1, tx.type, output); 
```

wouldnt that work?

Reply:
Compact and nice code. I applied the code with some modifications.
Thanks.